### PR TITLE
Add Refresh access token on expiration for sync session support

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

**Comment by bmunkholm** [comment]
@jsflax Shouldn't this target v10 branch?

**Comment by jsflax** [comment]
@bmunkholm I need James's changes for this and didn't want to step on his work.

File: tests/sync/app.cpp [hunk 1]
```diff
@@ -1300,9 +1327,21 @@ TEST_CASE("app: remote mongo client", "[sync][app]") {
...
             CHECK(breed == "king charles");
+        });
+        
+        dog_collection.find_one_and_update({{"name", "invalid name"}}, {{"name", "some name"}}, [&](Optional<bson::BsonDocument> document, Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(!document);
             processed = true;
         });
         
+//        dog_collection.find_one_and_update({{"name", "invalid name"}}, {{}}, find_and_modify_options, [&](Optional<bson::BsonDocument> document, Optional<app::AppError> error) {
```
Comment (kneth) [inline tests/sync/app.cpp:1338]: Remove if not needed

Comment (jsflax) [reply tests/sync/app.cpp:1338]: I believe this test is being commented out like the above due to a bug server side.

Comment (kneth) [reply tests/sync/app.cpp:1338]: Maybe a `FIXME: enable once server bug is fixed` is fine then :-)

**Pull request merged**